### ISSUE TITLE
BUG: User's default dataset not getting version update unless they click "change"

### ISSUE THREAD

**Issue by corneliusroemer** [nextstrain/nextclade#1139]
TL;DR: If you're a Nextclade user and seem to not get a new dataset version, please click "Change" and reselect that datset. That should fix it (until we've patched the bug).

We've had at least 3 reports of users getting old datasets after new datasets were released.

Based on user feedback, my assumption was thus far that users were using the same session, not reloading or restarting their machines. However, in retrospect this is quite unlikely given that one user reported having a dataset that was >2 months out of date (they will have most likely restarted their machine in between).

@emmahodcroft managed to catch the bug (getting old dataset) and we recorded what happened. In her case, opening a new tab she was shown an old dataset version (March instead of April)

Turns out that the index_v2.json _is_ loaded correctly (not cached), but it seems that we store the user's selected dataset across tabs. And that dataset is **pinned** to a particular **dataset version** not just **dataset type**.

A workaround for users is to simply click the "change" button which will unset the preference variables and pull in new version information from the index.

I never encountered this bug in my testing because I change datasets all the time. It's hard to reproduce for a frequent user. Thanks to all the users who reported: @JosetteSchoenma, Pavitra Roychoudhury and Emily Smith

Screenshot from Emma opening Nextclade in fresh tab:
<img width="1143" alt="image" src="https://user-images.githubusercontent.com/25161793/234619476-5395861d-4a3d-48af-acc7-cc2c2db56a59.png">

Emma has a detailed browser session recording available if required for debugging @ivan-aksamentov
